### ISSUE TITLE
insert JsonColumn using preparated statement

### ISSUE THREAD

**Issue by wi101** [outworkers/phantom#703]
using 2.11.1, the JsonColumn does not work with preparated statement. This is an Example:
```
case class User(id: Int, name: String)
case class Item (id: Int, name: String, description: Option[String], user: User)
```
```
abstract class Schema extends Table[Schema, Item] {

  object id extends IntColumn with PartitionKey
  object name extends StringColumn
  object description extends OptionalStringColumn
  object user extends JsonColumn[User] {
    override def fromJson(obj: String): User =
      decode[User](obj).fold(e => throw new Exception(e), identity)
    override def toJson(obj: User): String = obj.asJson.noSpaces
  }
}
```
```
class DB(val keyspace: CassandraConnection) extends Database[DB](keyspace) {

  case object items extends Schema with Connector {
    override def tableName: String = "items"

    def getItems: Future[List[Item]] = select.fetch()

    lazy val preparedInsert= insert
      .p_value(_.id, ?)
      .p_value(_.name, ?)
      .p_value(_.description, ?)
      .p_value(_.user, ?)
      .prepare()

    def insertItem(item: Item) =
      preparedInsert.bind(item.id, item.name, item.description, item.user)
        .future()
  }

}
```
 I get this compile error: 
```
 could not find implicit value for parameter binder: com.outworkers.phantom.macros.BindHelper[(Int, String, Option[String], example.product.model.User)]
[error]       preparedInsert.bind(item.id, item.name, item.description, item.user)
```

**Comment by alexflav23** [comment]
Hi @wi101,

This is now fixed in 2.12.0, which will be published today. Json columns are changing, you will no longer provide a `fromJson` and `toJson` method, instead that will be defined in a derived json Primitive.
```scala
case class User(id: Int, name: String)

object User {
  implicit val jsonPrimitive: Primitive[User] = {
    Primitive.json[User](obj => obj.asJson.noSpaces)(str => decode[User](str).fold(e => throw new Exception(e), identity))
  }
}

case class Item (id: Int, name: String, description: Option[String], user: User)
abstract class Schema extends Table[Schema, Item] {

  object id extends IntColumn with PartitionKey
  object name extends StringColumn
  object description extends OptionalStringColumn
  object user extends JsonColumn[User]
}
```

`JsonColumn` and the whole lot are now nothing more than type aliases, and the actual implementation has been completely removed from the framework. This was done because the macros rely on implicit primitives throughout, and implementing special behaviour would've been time consuming and not beneficial, since it's better to have a streamlined approach.

Regards.

**Comment by alexflav23** [comment]
Fixed in `2.12.0`, live in 2 hours on Maven Central.
